WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListView.swift

```diff
@@ -0,0 +1,234 @@
+                ForEach(viewModel.recentSites, id: \.domain) { site in
+                    siteButton(site: site)
+                }
+            } header: {
+                sectionHeader(
+                    title: Strings.recentsSectionTitle
+                )
+                .listRowInsets(Constants.sectionInsets)
+            }
+        }
+    }
+
+    @ViewBuilder
+    private func siteButton(site: Site) -> some View {
+        if let siteID = site.id {
+            Button {
+                if isEditing {
+                    withAnimation {
+                        viewModel.togglePinnedSite(siteID: siteID)
+                    }
+                } else {
+                    viewModel.siteSelected(siteID: siteID)
+                    selectionCallback(siteID)
+                }
+            } label: {
+                siteHStack(site: site)
+            }
+            .listRowSeparator(.hidden)
+            .buttonStyle(SelectedButtonStyle(onPress: { isPressed in
+                pressedDomains = pressedDomains.symmetricDifference([site.domain])
+            }))
+            .listRowBackground(
+                pressedDomains.contains(
+                    site.domain
+                ) ? Color.DS.Background.secondary : Color.DS.Background.primary
+            )
+        }
+    }
+
+    private func siteHStack(site: Site) -> some View {
+        HStack(spacing: 0) {
+            AvatarsView(style: .single(site.imageURL))
+                .padding(.trailing, .DS.Padding.split)
+
+            textsVStack(title: site.title, domain: site.domain)
+
+            Spacer()
+
+            if isEditing {
+                pinIcon(site: site)
+                    .padding(.leading, .DS.Padding.single)
+            }
+        }
+    }
+
+    private func textsVStack(title: String, domain: String) -> some View {
+        VStack(alignment: .leading, spacing: 0) {
+            Text(title)
+                .style(.bodySmall(.regular))
+                .foregroundStyle(Color.DS.Foreground.primary)
+                .layoutPriority(1)
+                .lineLimit(2)
+
+            Text(domain)
+                .style(.bodySmall(.regular))
+                .foregroundStyle(Color.DS.Foreground.secondary)
+                .layoutPriority(2)
+                .lineLimit(1)
+                .padding(.top, .DS.Padding.half)
+        }
+    }
+
+    private func pinIcon(site: Site) -> some View {
+        if viewModel.pinnedSites.contains(site) {
+            Image(systemName: "pin.fill")
+                .foregroundStyle(Color.DS.Background.brand(isJetpack: true))
+                .rotationEffect(.degrees(45))
+        } else {
+            Image(systemName: "pin")
+                .foregroundStyle(Color.DS.Foreground.secondary)
+                .rotationEffect(.degrees(45))
+        }
+    }
+
+}
+
+private extension BlogListView {
+    enum Strings {
+        static let pinnedSectionTitle = NSLocalizedString(
+            "site_switcher_pinned_section_title",
```

Comment:
Nit: The localized key typically follows the reverse DNS format:

```suggestion
            "site_switcher.pinned_section.title",
```